enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,845 @@
+status sub-resource, to simplify the UX of the admin user.
+
+An alternative would be to reference the UDN by the NAD name - with that
+information, whatever controller reconciling the `IPPool` CRs can access the
+NAD, and extract the network name. This approach would require a validating
+webhook to ensure that you can't have multiple `IPPool`s referencing the same
+logical network (since multiple NADs with the same logical network name - in
+**different** namespaces can exist).
+
+##### Centralized IP management workflow
+
+The [centralized IP management](#centralized-ip-management) flow is described
+in the following sequence diagram:
+```mermaid
+sequenceDiagram
+actor Admin
+actor Project admin
+
+participant MTV
+participant CNV
+participant o as OVN-Kubernetes
+
+Admin ->> o: cluster UDN
+o -->> Admin: OK
+
+Admin ->> CNV: provision IPPool
+CNV -->> Admin: OK
+
+Project admin ->> MTV: import VM
+MTV ->> CNV: create VM(name=<...>, primaryUDNMac=origMAC)
+CNV ->> CNV: ips = getIPsForMAC(mac=origMAC)
+CNV ->> o: create pod(mac=origMAC, IPs=ips)
+o -->> CNV: OK
+CNV -->> MTV: OK
+MTV -->> Project admin: OK
+```
+
+Hence, the required changes would be:
+- ipam-extensions (CNV component) and OVN-Kubernetes API will need to be
+  changed, to work with the multus default network annotation. Depending on the
+  VM's specification, it will request a specific MAC and IP addresses for the
+  primary UDN attachment. The `ipam-claim-reference` will also be requested via
+  this annoation.
+- the `IPAMClaim` CRD will need to be updated, adding a `Conditions` array to
+  its status. This way we will be able to report errors back to the user (e.g.
+  the desired IP allocation is already in use)
+- new CRD to be added, where the MAC <-> IPs addresses association will be
+  persisted. Only required for the
+  [centralized IP management](#centralized-ip-management) option
+- ipam-extensions (CNV component) will now also read the `IPPool` CRs for VMs
+  having primary UDNs in their namespaces, and requesting a specific MAC address
+  in their specs. These CRs will be used to generate the multus default network
+  annotation, which will be set in the pods by the mutating webhook.
+
+On a second stage, MTV (or other source cluster introspection tool) will
+provision the `IPPool` CR for the UDN on behalf of the admin user, thus
+simplifying the operation of the solution, making it more robust and less
+error-prone.
+
+###### Admin user flows for centralized approach
+
+The admin user will have to perform the following operations:
+- provision the cluster UDN CR having the overlay definition.
+  - layer2 topology ; IPAM lifecycle `Persistent` are a **must**
+- provision the `IPPool` object, pointing at the logical network created in the
+  previous step via the `IPPool.Spec.NetworkName attribute`. This is only
+  required until another tool automatically provisions the `IPPool` on behalf of
+  the admin
```

Comment:
Please note the cluster UDN logical network name is not exposed in the CR, its visible on the corresponding NAD via annotation or the spec.config blob.
It require additional "effort" from the user to find the network-name manually.

Reply:
Tks for noting. 

This is pretty much an abandoned idea we've entertained. While certainly relevant, it should be discussed in a follow-up design, which would discussed an epic which does not even exist yet.